standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/security/TestTokenStoreDelegationTokenSecretManager.java

```diff
@@ -99,19 +112,46 @@ private DelegationTokenIdentifier getID(String tokenStr) throws IOException {
 
   @Test public void testExpiry() throws IOException, InterruptedException {
     DelegationTokenStore tokenStore = new MemoryTokenStore();
-    TokenStoreDelegationTokenSecretManager mgr = createTokenMgr(tokenStore, 1);
+    TokenStoreDelegationTokenSecretManager mgr = createTokenMgr(tokenStore, 1, 8, 2);
     try {
       mgr.startThreads();
       String tokenStr =
           mgr.getDelegationToken(UserGroupInformation.getCurrentUser().getShortUserName(),
               UserGroupInformation.getCurrentUser().getShortUserName());
       DelegationTokenIdentifier id = getID(tokenStr);
       Assert.assertNotNull(mgr.verifyDelegationToken(tokenStr));
-      // Sleep for the renewal duration
-      Thread.sleep(1000);
-      SecretManager.InvalidToken ex = Assert.assertThrows(SecretManager.InvalidToken.class,
+      // Sleep for the expiration/maxlife duration duration
+      Thread.sleep(6000);
+      IllegalStateException ex = Assert.assertThrows(IllegalStateException.class,
           () -> mgr.verifyDelegationToken(tokenStr));
-      Assert.assertTrue(ex.getMessage(), ex.getMessage().contains("has expired"));
+      Assert.assertTrue(ex.getMessage(), ex.getMessage().contains("Expiration time passed"));
+      Thread.sleep(6000);
+      //Expiry thread will remove the token as it has crossed the maxLifeTime.
+      Assert.assertEquals(tokenStore.getAllDelegationTokenIdentifiers().size(), 0);
+    } finally {
+      mgr.stopThreads();
+    }
+  }
+
+  @Test
+  public void testExpiryThreadRenewingAndRemovingToken() throws IOException, InterruptedException {
+    DelegationTokenStore tokenStore = new MemoryTokenStore();
```

Comment:
can we add this case on the above testcase itself

Reply:
I have kept 2 tests only as both are doing different functionality, but reduced the sleep time by a lot.